src/main/webapp/user_profile_script.js

```diff
@@ -83,26 +90,101 @@ async function refresh() {
-        var msg = msgResponse[index];
-        var newMessage = document.createElement("div");
+    if (msgResponse.length === 10) {
+        let msg = msgResponse[9];
+        let newMessage = document.createElement("div");
         newMessage.className = msg.className;
         newMessage.appendChild(document.createTextNode(msg.message)); 
+        let moreMessageBtn = document.createElement("button");
+        moreMessageBtn.setAttribute("class", "more-msg-button");
+        moreMessageBtn.textContent = "Load 10 more messages";
+        moreMessageBtn.addEventListener("click", function() { 
+            loadMoreMessages(keyString, newMessage)
+        });
+        msgContainer.appendChild(moreMessageBtn);
         msgContainer.appendChild(newMessage);
+        for (var index = 8; index >= 0; index--) {
+            var message = msgResponse[index];
+            var newMessageDiv = document.createElement("div");
+            newMessageDiv.className = message.className;
+            newMessageDiv.appendChild(document.createTextNode(message.message)); 
+            msgContainer.appendChild(newMessageDiv);
+        }
+    } else {
+        for (var index = msgResponse.length - 1; index >= 0; index--) {
+            var msg = msgResponse[index];
+            var newMessage = document.createElement("div");
+            newMessage.className = msg.className;
+            newMessage.appendChild(document.createTextNode(msg.message)); 
+            msgContainer.appendChild(newMessage);
+        }
     }
 
     // Keep message container scrolled to bottom at the beginning
     msgContainer.scrollTop = msgContainer.scrollHeight;
 }
 
+// loadMoreMessages(key, messageDiv) will load 10 more messages of
+// task with keyString equals to key. After loading, scroll to element messageDiv
+async function loadMoreMessages(key, messageDiv) {
+    const queryURL = "/messages?key=" + key + "&cursor=" + cursorString;
+    const request = new Request(queryURL, {method: "GET"});
+    const response = await fetch(request);
+    const result = await response.json();
+    const msgResponse = result.messages;
+    cursorString = result.cursorString;
+
+    const msgContainer = document.getElementById("message-container");
+    msgContainer.removeChild(msgContainer.childNodes[0]);
+    if (msgResponse.length === 10) {
+        var newMessageDiv;
+        for (var index = 0; index < 10; index++) {
+            var message = msgResponse[index];
+            newMessageDiv = document.createElement("div");
+            newMessageDiv.className = message.className;
+            newMessageDiv.appendChild(document.createTextNode(message.message)); 
+            msgContainer.insertBefore(newMessageDiv, msgContainer.childNodes[0]);
+        }
+        var moreMessageBtn = document.createElement("button");
+        moreMessageBtn.setAttribute("class", "more-msg-button");
+        moreMessageBtn.textContent = "Load 10 more messages";
```

Comment:
I'd reword this to 'Load 10 previous messages' or Load older messages

Reply:
Sure. "Load 10 previous messages" is better than my present wording and I would make the change in a new PR.